src/NexusMods.App/AppConfig.cs

```diff
@@ -96,28 +97,52 @@ public class LoggingSettings : ILoggingSettings
     /// <summary>
     /// Default constructor for serialization.
     /// </summary>
-    public LoggingSettings() : this(FileSystem.Shared) {}
+    public LoggingSettings() : this(FileSystem.Shared) { }
 
     /// <summary>
     /// Creates the default logger with logs stored in the entry directory.
     /// </summary>
     /// <param name="fileSystem">The FileSystem implementation to use.</param>
     public LoggingSettings(IFileSystem fileSystem)
     {
-        var baseFolder = fileSystem.GetKnownPath(KnownPath.EntryDirectory);
-        baseFolder.CreateDirectory();
-        FilePath = new ConfigurationPath(baseFolder.Combine(LogFileName));
-        ArchiveFilePath = new ConfigurationPath(baseFolder.Combine(LogFileNameTemplate));
+        var baseFolder = GetDefaultBaseDirectory(fileSystem);
+        FilePath = GetFilePath(baseFolder);
+        ArchiveFilePath = GetArchiveFilePath(baseFolder);
         MaxArchivedFiles = 10;
     }
 
     /// <summary>
     /// Expands any user provided paths; and ensures default settings in case of placeholders.
     /// </summary>
-    public void Sanitize()
+    public void Sanitize(IFileSystem fs)
     {
         MaxArchivedFiles = MaxArchivedFiles < 0 ? 10 : MaxArchivedFiles;
+
+        // Set default locations if none are provided.
+        var baseFolder = GetDefaultBaseDirectory(fs);
+        if (string.IsNullOrEmpty(FilePath.RawPath))
+            FilePath = GetFilePath(baseFolder);
+
+        if (string.IsNullOrEmpty(ArchiveFilePath.RawPath))
+            ArchiveFilePath = GetArchiveFilePath(baseFolder);
+
+        FilePath.ToAbsolutePath().Parent.CreateDirectory();
+        ArchiveFilePath.ToAbsolutePath().Parent.CreateDirectory();
+    }
+
+    private static AbsolutePath GetDefaultBaseDirectory(IFileSystem fs)
+    {
+        return fs.OS.MatchPlatform(
+            () => fs.GetKnownPath(KnownPath.LocalApplicationDataDirectory).Combine("NexusMods.App/Logs"),
+            () => fs.GetKnownPath(KnownPath.XDG_DATA_HOME).Combine("NexusMods.App/Logs"),
+            () => throw new NotSupportedException(
```

Comment:
Logs should go into the intended XDG base directory: `XDG_STATE_HOME` as defined by the [specs](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).